ee/EAP7-1114_JakartaEE8.adoc

```diff
@@ -0,0 +1,42 @@
+= Pass the Jakarta EE 8 certification testsuite
+:author:            Scott Marlow
+:email:             smarlow@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+We will run and pass the Jakarta EE 8 CTS with WildFly and our EAP product.  This will be done with https://github.com/eclipse-ee4j/jakartaee-tck.  FYI, the TCK development mailing list is https://www.eclipse.org/lists/jakartaee-tck-dev/maillist.html.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/EAP7-1114 Jakarta EE 8 Certification
+
+=== Dev Contacts
+
+* mailto:{Scott Marlow}[{smarlow@redhat.com}] JPA / WildFly integration lead
+
+=== QE Contacts
+
+mailto:{Martin Simka}[{msimka@redhat.com}]
+
+=== Affected Projects or Components
+
+Since WildFly is already Java EE 8 certified, in theory, no projects or components should be impacted by this work.  
+
+=== Known Blockers
+
+* Missing jars https://github.com/eclipse-ee4j/jakartaee-tck/issues/4
+* Hardcoded references to JDK and property file https://github.com/eclipse-ee4j/jakartaee-tck/issues/7
+
+=== Hard Requirements
+
+Pass the https://github.com/eclipse-ee4j/jakartaee-tck CTS.
+
+== Test Plan
+
+Build/obtain the Jakarta EE 8 CTS and pass it with WildFly + EAP.
```

Comment:
When we develop the new configuration do we want to consider testing with Elytron by default this time and PicketBox secondary?  Within the next year we plan to be removing PicketBox so if we already default to Elytron it will be the secondary configuration that needs removing.

Reply:
We still don't have the fix for JBCTS-1436 (The deployment for testSAMObtainBean includes the permissions.xml at the wrong location).  Would you workaround passing the testSAMObtainBean or wait for the proper fix?

Reply:
At the moment I don't know what our integration is going to look like, are we going to use what we used before or take the opportunity to rewrite our integration to use the CLI rather than configuration files?

For this specific issue, if a fix is not available yet we can always clone the permissions.xml descriptors and rename them jboss-permissions.xml - this mis-interpretation of the spec is why we added our own form of the descriptor to allow the permissions to be overridden at the child deployment level.

However if it was possible for me to run the TCK myself I would likely send over a TCK PR myself to correct the location of this file.

But really this is also why I want to be able to get the TCK running locally as soon as possible, very subtle innocent looking changes due to clean up in the TCK itself could have pretty massive impacts on our test results so I want to be following the security changes as they happen.

Reply:
The Jakarta EE 8 CTS is supposed to be the same as Java EE 8 CTS, meaning that we shouldn't need to change anything to pass Jakarta EE 8 CTS.  Our goal is to pass Jakarta EE 8 CTS without any changes, as soon as possible (e.g. for the current EAP release).  

As per [Jakartaee-tck README.md](https://github.com/eclipse-ee4j/jakartaee-tck/blob/master/README.md), the CTS is not ready to be built/used yet.  More specifically, they asked everyone to stop trying to build it (I tried and posted on the [jakarta tck dev ml).](https://dev.eclipse.org/mailman/listinfo/jakartaee-tck-dev)

Having said that, once you/everyone can run the TCK, it wouldn't hurt to create a PR for JBCTS-1436, so that others know about the problem with that test and the fix can be planned (e.g. merge for Jakarta EE 8 or 9).

Regarding TCK integration/configuration, using the CLI is a long term goal that we haven't worked on yet (due to lack of volunteers).  For WF15/WF16 you should plan to keep using the static configuration.  Up to you, whether you do your configuration changes on a [cts-8-mods branch ](https://gitlab.cee.redhat.com/j2eects/cts-8-mods) or in a new file that is a copy of [standalone-full-with-elytron.xml](https://gitlab.cee.redhat.com/j2eects/cts-8-mods/blob/master/etc/standalone-full-with-elytron.xml).

The conversion to use CLI, needs to be on separate branches, so that master (TCK) branches are not impacted, until the task is completed.  FYI @maeste @tomjenkinson jira is [JBCTS-1352](https://issues.jboss.org/browse/JBCTS-1352).